app/services/trainees/create_from_dttp.rb

```diff
@@ -0,0 +1,333 @@
+
+      { locale_code: Trainee.locale_codes[:uk] }
+    end
+
+    def contact_attributes
+      {
+        address_line_one: dttp_trainee.response["address1_line1"],
+        address_line_two: dttp_trainee.response["address1_line2"],
+        town_city: dttp_trainee.response["address1_line3"],
+        postcode: dttp_trainee.response["address1_postalcode"],
+        email: dttp_trainee.response["emailaddress1"],
+      }.merge(locale_code)
+    end
+
+    def course_attributes
+      {
+        course_subject_one: course(latest_placement_assignment.response["_dfe_ittsubject1id_value"]),
+        course_subject_two: course(latest_placement_assignment.response["_dfe_ittsubject2id_value"]),
+        course_subject_three: course(latest_placement_assignment.response["_dfe_ittsubject3id_value"]),
+        course_min_age: age_range && age_range[0],
+        course_max_age: age_range && age_range[1],
+        study_mode: study_mode,
+        commencement_date: earliest_placement_assignment.response["dfe_commencementdate"],
+        course_start_date: latest_placement_assignment.response["dfe_programmestartdate"],
+        course_end_date: latest_placement_assignment.response["dfe_programmeeenddate"],
+      }
+    end
+
+    def course(dttp_course_uuid)
+      find_by_entity_id(
+        dttp_course_uuid,
+        Dttp::CodeSets::CourseSubjects::MAPPING,
+      )
+    end
+
+    def age_range
+      @age_range ||= find_by_entity_id(
+        latest_placement_assignment.response["_dfe_coursephaseid_value"],
+        Dttp::CodeSets::AgeRanges::MAPPING,
+      )
+    end
+
+    def study_mode
+      find_by_entity_id(
+        latest_placement_assignment.response["_dfe_studymodeid_value"],
+        Dttp::CodeSets::CourseStudyModes::MAPPING,
+      )
+    end
+
+    def school_attributes
+      return {} if latest_placement_assignment.lead_school_id.blank?
+
+      # Should we raise when schools are not found so that we can add them?
+      attrs = {
+        lead_school: School.find_by(urn: lead_school_urn),
+      }
+
+      if latest_placement_assignment.employing_school_id.present?
+        attrs.merge!({
+          employing_school: School.find_by(urn: employing_school_urn),
+        })
+      end
+
+      attrs
+    end
+
+    def lead_school_urn
+      Dttp::School.find_by(dttp_id: latest_placement_assignment.lead_school_id)&.urn
+    end
+
+    def employing_school_urn
+      Dttp::School.find_by(dttp_id: latest_placement_assignment.employing_school_id)&.urn
+    end
+
+    def funding_attributes
+      return {} unless latest_placement_assignment.response["dfe_allocatedplace"] == Dttp::Params::PlacementAssignment::ALLOCATED_PLACE
+
+      if latest_placement_assignment.response["_dfe_bursarydetailsid_value"] == Dttp::Params::PlacementAssignment::SCHOLARSHIP
+        return { applying_for_scholarship: true }
+      end
+
+      funding_method = FundingMethod.find_by(training_route: training_route_for_funding)
```

Comment:
I don't think the tiered bursaries are in our database, so this won't pick them up.

Tiered bursaries might need some thinking about because I'm not sure we're handling them in a 'yearly' way anywhere yet..

Reply:
Added a fix in https://github.com/DFE-Digital/register-trainee-teachers/pull/1787/commits/0f0b5bae9ae304153542a2cd48130406369d3632
Not quite handling (academic) yearly bit yet, but only as it stands in register/dttp at the moment.